### PR TITLE
python3Packages.django-oauth-toolkit: 2.4.0 -> 3.0.1

### PR DESCRIPTION
**Pull request by onny**
Diff: https://github.com/jazzband/django-oauth-toolkit/compare/2.4.0...3.0.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by natsukium** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353483`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>baserow</li>
    <li>netbox_3_7</li>
    <li>netbox_4_0</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pretix</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>netbox (netbox_4_1)</li>
    <li>peering-manager</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.netbox-bgp</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-reorder-rack</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>baserow</li>
    <li>netbox_3_7</li>
    <li>netbox_4_0</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pretix</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>netbox (netbox_4_1)</li>
    <li>peering-manager</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.netbox-bgp</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-reorder-rack</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>baserow</li>
    <li>netbox_3_7</li>
    <li>netbox_4_0</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/348af5527826a49ac5a2ee37b67f56ff">tandoor-recipes</a></li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>netbox (netbox_4_1)</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.netbox-bgp</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-reorder-rack</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-reorder-rack</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>baserow</li>
    <li>netbox_3_7</li>
    <li>netbox_4_0</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/3c466e7f173e9cde1a4221d2b4c80f41">tandoor-recipes</a></li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>netbox (netbox_4_1)</li>
    <li>python311Packages.django-oauth-toolkit</li>
    <li>python311Packages.drf-spectacular</li>
    <li>python311Packages.netbox-bgp</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-reorder-rack</li>
    <li>python312Packages.django-oauth-toolkit</li>
    <li>python312Packages.drf-spectacular</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-reorder-rack</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
`tandoor-recipes` should be fixed with https://github.com/NixOS/nixpkgs/pull/354125

**Comment by mweinelt** [comment]
Ping me on pretix breakages. I care and I react quickly :shrug:

File: pkgs/development/python-modules/django-oauth-toolkit/default.nix [hunk 1]
```diff
@@ -18,29 +19,29 @@
...
+    repo = "django-oauth-toolkit";
     rev = "refs/tags/${version}";
-    hash = "sha256-nfLjjVp+6OsjFdJHUZ2gzZic/E/sCklj+YeFyb/EZdw=";
+    hash = "sha256-Ya0KlX+vtLXN2Fgk0Gv7KemJCUTwkaH+4GQA1ByUlBY=";
   };
 
   postPatch = ''
-    sed -i '/cov/d' tox.ini
+    sed -i '/cov/d' pyproject.toml
   '';
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-oauth-toolkit/default.nix:34]: This should have used the pytest-cov-stub instead.

**Pull request merged**